### PR TITLE
PAT Public API

### PR DESCRIPTION
**Pull request by nmartin413**
Updates PAT to use either the lambda interface or a new graphql interface.

- refactors the calls into a generic interface
- consolidates the lambda calls
- adds graphql support (and new arguments for API host and token)
- adds a connection check method (Public API only)
- removes unused code

### CODE REVIEW

**Review by darwayne — COMMENTED** [review]
🤩 looking good so far

File: panther_analysis_tool/backend/public_api_client.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    def delete_detections(self, params: DeleteDetectionsParams) -> BackendResponse:
+        pass
+
+    def _execute(self, request: DocumentNode) -> ExecutionResult:
+        return self._gql_client.execute(request, get_execution_result=True)
+
+
+_API_URL_PATH = "public/graphql"
+_API_DOMAIN_PREFIX = "api"
+_API_TOKEN_HEADER = "X-API-Key"
```
Comment (darwayne) [inline panther_analysis_tool/backend/public_api_client.py:127]: I think you'll need to throw in a lint ignore flag here to appease the ci gods

**Review by danbiwer — APPROVED** [review]
Nice, LGTM.

I suppose we will also want to add `putSchema`and `listSchemas`.  I think you actually had those in at some point before I said PAT wasn't using them 😬

**Comment by kostaspap** [comment]
> Nice, LGTM.
> 
> I suppose we will also want to add `putSchema`and `listSchemas`. I think you actually had those in at some point before I said PAT wasn't using them 😬

Yup, we have customer using PAT to manage their custom schemas!

**Review by danbiwer — APPROVED** [review]
awesome, huge improvement

**Review by danbiwer — APPROVED** [review]

**Review by darwayne — APPROVED** [review]

**Pull request merged**